src/connection.rs

```diff
@@ -30,6 +48,19 @@ impl Connection {
                 "memcache URL should start with 'memcache://'".into(),
             ));
         }
+        if connection_type == (ConnectionType::UDP as u32) {
+            let socket = UdpSocket::bind("0.0.0.0:0")?;
+            socket.connect(addr.clone())?;
+            return Ok(Connection {
+                stream: Stream::UdpSocket(UdpConnection {
+                    socket: socket,
+                    read_buf: Vec::new(),
+                    write_buf: Vec::new(),
+                    request_id:  rand::random::<u16>()
```

Comment:
I think request id can be same across connections and only needs to be different for multiple requests in the same connection,  and since a single udpconnection processes one request at a time so this should not be a problem.